Scripts/Game/ODIN/Editor/Components/EditableEntity/SCR_EditableGroupComponent.c

```diff
@@ -10,21 +10,12 @@ modded class SCR_EditableGroupComponent : SCR_EditableEntityComponent
 			
 		if (waypointCompToRemove && waypointCompToRemove.GetParentEntity() == this)
 		{
-			// Only delete if there is no waypoint cycle present
-			bool has_cycle = false;
-			array<AIWaypoint> waypoints = {};
-			m_Group.GetWaypoints(waypoints);
+			bool has_cycle = AIWaypointCycle.Cast(wp);
 			
-			foreach (AIWaypoint waypoint: waypoints)
-			{
+			if (!has_cycle)
+				has_cycle = ODIN_HasWaypointCycle();
```

Comment:
Any particular reason you check first for this specific WP and then for all waypoints, if it isn't a cycle? 
Or is it purely "optimizing" to avoid checking every waypoint, if the current one is a cycle? 
As the ``ODIN_HasWaypointCycle()`` would return the same result even if current waypoint is a cycle?

Reply:
Checking the current waypoint is the fix. In Reforger 0.9, the current waypoint was also returned by `AIGroup.GetWaypoints`. In 1.0 it isn't, which caused the cycle waypoint getting deleted.

Reply:
ahhh gotcha. I didn't catch the part about it not returning the current waypoint in 1.0. That almost sounds like a regression as I would expect `AIGroup.GetWaypoints` to return all waypoints in the list. But as you said, might be a race condition with the completion/remove event. In any case your fix handles either situation.